library/core/src/prelude/rust_2024.rs

```diff
@@ -0,0 +1,100 @@
+//! The 2024 version of the core prelude.
```

Comment:
Instead of duplication, would it make sense to define a "base prelude" module with items that are in every prelude form every edition?

Reply:
That is infinitely better…I will definitely do this.

Reply:
Done.